app/src/main/java/com/github/orkest/shazam/data/AudioRecording.kt

```diff
@@ -0,0 +1,135 @@
+
+
+class AudioRecording {
+
+    companion object {
+
+       @RequiresPermission(Manifest.permission.RECORD_AUDIO)
+       private fun buildAudioRecord(): AudioRecord {
+           //Set the audio source
+           val audioSource = MediaRecorder.AudioSource.UNPROCESSED
+
+           //Set the audio format
+           val audioFormat = AudioFormat.Builder()
+               .setChannelMask(AudioFormat.CHANNEL_IN_MONO)
+               .setEncoding(AudioFormat.ENCODING_PCM_16BIT)
+               .setSampleRate(48_000)
+               .build()
+
+           //Create and return the audio recording object based on the source and format
+           return AudioRecord.Builder()
+               .setAudioSource(audioSource)
+               .setAudioFormat(audioFormat)
+               .build()
+       }
+
+        private fun buildReadingBuffer(): ByteArray {
+            // size of buffer to retrieve chunks of audio
+            val bufferSize = AudioRecord.getMinBufferSize(
+                ShazamConstants.RECORDING_SAMPLE_RATE,
+                AudioFormat.CHANNEL_IN_MONO,
+                AudioFormat.ENCODING_PCM_16BIT
+            )
+            // Buffer to store retrieved chunks of audio
+            return ByteArray(bufferSize)
+        }
+
+        /**
+         * Records AudioChunks continuously and returns a [Flow] of [AudioChunk]s
+         * Must be called in the coroutineScope as the streaming session
+         * @throws [SecurityException] if the app does not have the RECORD_AUDIO permission
+         */
+        @RequiresPermission(Manifest.permission.RECORD_AUDIO)
+        fun recordingFlow(coroutineScope: CoroutineScope): Flow<AudioChunk> {
+            //Build the audioRecord object
+            val audioRecord = buildAudioRecord()
+
+            //Check if the audioRecord is initialized correctly
+            require(audioRecord.state == AudioRecord.STATE_INITIALIZED) {
+                Log.d("AudioRecord","AudioRecord is not initialized")
+            }
+
+            //Build the buffer to store the audio chunks from the audioRecord (microphone)
+            val readBuffer = buildReadingBuffer()
+
+            //Return the flow of AudioChunks, will be actually run in a suspending way when collected
+            return flow {
+                audioRecord.startRecording()
+                //Collect audio chunks while the coroutine is active
+                while (currentCoroutineContext().isActive) {
+                    val actualRead = audioRecord.read(readBuffer, 0, readBuffer.size)
+                    val byteArray = readBuffer.sliceArray(0 until actualRead)
+                    val audioChunk = AudioChunk(byteArray, actualRead, System.currentTimeMillis())
+                    emit(audioChunk)
+                }
+                //Stop the audioRecord when the coroutine is cancelled
+                // (either because of result, or user stopped recording)
+                audioRecord.release()
```

Comment:
Where do you handle the case in which a result is obtained?

Reply:
In this PR I only coded the audio recording, the audio recognition part will be the object of my next PR. The result will be handled at that moment.